third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -916,16 +914,33 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                         specifier.loc,
                         EA::ModuleAccess_::ModuleAccess(mident, *resource),
                     );
-                    let ety = sp(
-                        specifier.loc,
-                        EA::Type_::Apply(maccess, type_args.as_ref().cloned().unwrap_or_default()),
-                    );
-                    let ty = self.translate_type(&ety);
-                    if let Type::Struct(mid, sid, inst) = ty {
-                        ResourceSpecifier::Resource(mid.qualified_inst(sid, inst))
+                    let sym = self.parent.module_access_to_qualified(&maccess);
+                    if let Type::Struct(mid, sid, _) = self.parent.parent.lookup_type(&loc, &sym) {
+                        if type_args.is_none() {
+                            // If no type args are provided, we assume this is a partial type without
```

Comment:
This comment is puzzling.  What if the type is not a generic type?  Wouldn't there also be no type args provided in that case?

Reply:
Everything is a generic type, some of them have zero parameters, but done.